### PR TITLE
[autoscaler] cover autofilling resources with try except.

### PR DESCRIPTION
**Pull request by AmeerHajAli**
solves #12195.
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/_private/commands.py [hunk 1]
```diff
@@ -270,8 +270,10 @@ def _bootstrap_config(config: Dict[str, Any],
 
     cli_logger.print("Checking {} environment settings",
                      _PROVIDER_PRETTY_NAMES.get(config["provider"]["type"]))
-
-    config = provider_cls.fillout_available_node_types_resources(config)
+    try:
+        config = provider_cls.fillout_available_node_types_resources(config)
+    except Exception:
+        logger.exception("Failed to autodetect node resources.")
```
Comment (richardliaw) [inline python/ray/autoscaler/_private/commands.py:276]: can you please do:

```
except Exception as exc:
if cli_logger.verbosity > 2:
    logger.exception(...)
else:
    cli_logger.warning(f"Failed to autodetect node resources: {str(exc)}. "
                                    "See full stack trace with higher verbosity."
```

**Review by richardliaw — CHANGES_REQUESTED** [review]
Thanks @AmeerHajAli for the fast turnaround. Can you please make the requested change, so that the _first-time user_ does not need to see this scary stacktrace?

**Comment by AmeerHajAli** [comment]
I made the changes.
PS: author-action-required label might speedup the process.

**Comment by richardliaw** [comment]
Thanks! i was asleep during that time anyways :)

**Review by richardliaw — APPROVED** [review]
Thanks for the fast turnaround, Ameer!

**Pull request merged**